x-pack/plugins/security_solution/public/onboarding/components/onboarding_body/cards/common/card_selector_list.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useEffect } from 'react';
+import { EuiPanel, EuiFlexGroup, EuiFlexItem, EuiTitle, EuiText, EuiSpacer } from '@elastic/eui';
+import type { RulesCardItemId } from '../rules/types';
+import type { AlertsCardItemId } from '../alerts/types';
+import type { DashboardsCardItemId } from '../dashboards/types';
+import { useCardSelectorListStyles } from './card_selector_list.styles';
+
+const INITIAL_SCROLL_ANIMATION_DURATION = 500;
+const SCROLL_ANIMATION_DURATION = 300;
+
+export enum CardSelectorListItemAssetType {
+  video = 'video',
+  image = 'image',
+}
+export interface CardSelectorListItem {
+  id: RulesCardItemId | AlertsCardItemId | DashboardsCardItemId;
+  title: string;
+  description: string;
+  asset: {
+    type: CardSelectorListItemAssetType;
+    source: string;
+    alt: string;
+  };
+}
+
+interface CardSelectorListProps {
+  items: CardSelectorListItem[];
+  onSelect: (item: CardSelectorListItem) => void;
+  selectedItem: CardSelectorListItem;
+  title?: string;
+}
+
+const scrollToSelectedItem = (
+  cardId: string,
+  animationDuration: number = SCROLL_ANIMATION_DURATION
+) => {
+  setTimeout(() => {
+    const element = document.getElementById(`selector-${cardId}`);
+    if (element) {
+      element.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+    }
+  }, animationDuration);
+};
+
+export const CardSelectorList = React.memo<CardSelectorListProps>(
+  ({ items, onSelect, selectedItem, title }) => {
+    const styles = useCardSelectorListStyles();
+
+    useEffect(() => {
+      scrollToSelectedItem(selectedItem.id, INITIAL_SCROLL_ANIMATION_DURATION);
+    }, [selectedItem]);
```

Comment:
It's somewhat annoying that the auto-scroll is executed when I manually click an item, manual actions should not trigger auto-scroll. Let's do it only in the first render:
```suggestion
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
```